cpp/src/connect/weak_cc.cuh

```diff
@@ -0,0 +1,343 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+///#include "cuda_utils.h"
+///#include "array/array.h"
+
+#include <thrust/device_ptr.h>
+#include <thrust/scan.h>
+
+#include <cuda_runtime.h>
+#include <stdio.h>
+
+#include <iostream>
+#include <iomanip>
+#include <type_traits>
+
+#include "utils.h"
+
+
+namespace MLCommon {
+
```

Comment:
Do I understand this is the code from cuML? If so, has CC completely moved to cugraph or is this a duplication?

Reply:
This is from cuML, yes. As discussed, it is a duplication of a code adapted from cuML.

Reply:
Would that make sense to document the main changes/differences (if there are any) so that anyone can easily integrate fixes/improvement from one copy to another?

Reply:
Yes. Will take care of that.